util/tls/tls_socket.cc

```diff
@@ -165,6 +165,22 @@ auto TlsSocket::Close() -> error_code {
   return next_sock_->Close();
 }
 
+class SpinCounter {
+ public:
+  explicit SpinCounter(size_t limit) : limit_(limit) {
```

Comment:
@romange The way I would properly implement this is by mimicking the interface of the type I am trying to collect stats about. So in our case that's an int, so I would introduce a wrapper class, overload the operators and I would collect the stats internally such that the variable itself can track the changes and do the printing (because now we need `SpinCounter.Check()` on a separate variable) but this seemed to be an overkill for this so I opted in for the simplest and smallest snippet/class utlity possible

Reply:
Also if we need it elsewhere, I am happy to generalize move it somewhere else :)